src/lib/y2storage/proposal/partition_creator.rb

```diff
@@ -158,6 +160,28 @@ def create_partition(planned_partition, free_space, primary)
         end
       end
 
+      # Use the single partition of an implicit partition table instead of creating a new one
+      #
+      # @raise [Y2Storage::NoMorePartitionSlotError] if the single implicit partition is
+      #   already in use.
+      #
+      # @param ptable [Y2Storage::PartitionTables::ImplicitPt]
+      # @return [Y2Storage::Partition] single implicit partition
+      def use_implicit_partition(ptable)
+        partition = ptable.partition
+        return partition unless implicit_partition_in_use?(partition)
+
+        raise NoMorePartitionSlotError
```

Comment:
Please add a string describing what the problem was when raising a relatively generic exception.